src/main/java/com/verde/market/domain/repository/ProductRepository.java

```diff
@@ -0,0 +1,15 @@
+package com.verde.market.domain.repository;
+
+import com.verde.market.domain.Product;
+
+import java.util.List;
+import java.util.Optional;
+
```

Comment:
This interface has too many methods. Probably, you should try to find the best way to split it according responsability or another parameter